Bio/SeqUtils/__init__.py

```diff
@@ -667,6 +667,46 @@ def calculate(self, sequence):
 
         return exp(cai_value / cai_length)
 
+    def optimize(self, sequence, seq_type="DNA", table=standard_dna_table):
```

Comment:
Would it be meaningful to use a different `table` than the one already passed to `CodonAdaptationIndex.__init__`? I don't think so. Instead, I'd suggest to add `self._table = table` to `__init__` and then use that here in the `optimize` method.

Reply:
Great point - will make change